pkg/v1/tkg/tkgpackageclient/package_update.go

```diff
@@ -16,10 +16,32 @@ import (
 
 	kappipkg "github.com/vmware-tanzu/carvel-kapp-controller/pkg/apis/packaging/v1alpha1"
 
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/log"
 	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/tkgpackagedatamodel"
 )
 
 func (p *pkgClient) UpdatePackage(o *tkgpackagedatamodel.PackageOptions, progress *tkgpackagedatamodel.PackageProgress, operationType tkgpackagedatamodel.OperationType) {
+	p.updatePackage(o, progress, operationType)
+}
+
+// UpdatePackageSync installs/updates the package and returns an error if any
+func (p *pkgClient) UpdatePackageSync(o *tkgpackagedatamodel.PackageOptions, operationType tkgpackagedatamodel.OperationType) error {
+	pp := getDefaultProcessConfig()
+
+	go p.updatePackage(o, pp, operationType)
+
+	initialMsg := fmt.Sprintf("Updating installed package '%s'", o.PkgInstallName)
+	if err := DisplayProgress(initialMsg, pp); err != nil {
+		if err.Error() == tkgpackagedatamodel.ErrPackageNotInstalled {
+			log.Warningf("package '%s' is not among the list of installed packages in namespace '%s'. Consider using the --install flag to install the package", o.PkgInstallName, o.Namespace)
```

Comment:
You probably want to remove the following part of the comment as there is no cmd layer involved in your case?
```
Consider using the --install flag to install the package
```